containers/datalab/run.sh

```diff
@@ -20,27 +20,68 @@
 # command prompt, which is useful for tinkering within the container before
 # manually starting the server.
 
+HERE=$(dirname $0)
 CONTENT=$HOME
 ENTRYPOINT="/datalab/run.sh"
-if [ "$1" != "" ]; then
-  if [ "$1" != "shell" ]; then
-    CONTENT=$1
-    shift
-  fi
-  if [ "$1" == "shell" ]; then
-    ENTRYPOINT="/bin/bash"
-  fi
-fi
+DEVROOT_DOCKER_OPTION=''
+while [ $# -gt 0 ]; do
+  case "$1" in
+    shell)
+      ENTRYPOINT="/bin/bash"
+      ;;
+    --live)
+      # Use --live to make the datalab container use a live copy of the
+      # development directory so your changes are visible to the container
+      # without rebuilding.
+      DEVROOT=$(cd "$HERE" && realpath ../..)
+      DEVROOT_DOCKER_OPTION="-v $DEVROOT:/devroot"
+      echo "Updating build-dev"
+      rsync -a \
+          --exclude web/nb/config \
+          --exclude web/nb/static \
+          --exclude web/nb/templates \
+          $DEVROOT/build/ $DEVROOT/build-dev
+      if [ ! -h $DEVROOT/build-dev/web/nb/node_modules ]; then
+        echo linking node_modules
+        ln -s /datalab/web/node_modules $DEVROOT/build-dev/web/nb/node_modules
```

Comment:
There are a whole bunch of symlinks in the build-dev directory. I don't think it's worth doing any partial clean up.